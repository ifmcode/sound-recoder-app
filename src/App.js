import Recorder from './components/Recorder'
import './App.less';
import { useEffect, useState, useMemo } from 'react';
import {Provider} from 'react-redux'
import store from './redux/store/store';

function App() {
  const constraints = useMemo(() => { return {audio: true} }, []),
    [stream, setStream] = useState(null),
    [error, setError] = useState(null);

  useEffect(() => {
    if ( stream ) {
      return
    }

    let didCancel = false

    const getUserMedia = async () => {
      try {
        if (!didCancel) {
          const stream = await navigator.mediaDevices.getUserMedia(constraints);
          setStream(stream);
        }
      } catch (err) {
        if (!didCancel) {
          setError(err);
        }
      }
    }

    const cancel = () => {
      didCancel = true;
      if (stream) {
        if ((stream).getAudioTracks) {
          (stream).getAudioTracks().map(track => track.stop());
        }
  
        if ((stream).stop) {
          (stream).stop();
        }
      }
    }

    getUserMedia();

    return cancel;
  }, [constraints, stream, error])
  
  return (
    <Provider store={store}>
      <header>
        <h1>Web Dictaphone</h1>
      </header>
      <main>
        {
          (stream)
            ? <Recorder stream={stream}/>
            : <p className="loading-phrase">Loadingâ€¦</p>
        }
      </main>
    </Provider>
  );
}

export default App;
