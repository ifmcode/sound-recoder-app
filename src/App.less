@import (reference) "./styles/colors.less";
@import (reference) "./styles/sizes.less";

div#root {
  width: @appWidth;
  height: @appHeight;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  background-color: @grey;
  margin: 0 auto;
  padding: 0 40px;
}

header {
  display: flex;
  width: 100%;
  justify-content: center;

  h1 {
    color: @bluegreen;
    margin: 30px 0;
  }
}

main {
  position: relative;
  width: 100%;
  justify-content: center;
  display: flex;
  flex-direction: column;

  .loading-phrase {
    color: @white;
    text-align: center;
  }
}
