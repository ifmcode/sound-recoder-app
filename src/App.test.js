import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
//import { render, screen } from '@testing-library/react';
import App from './App';

test('renders Sound Recorder title', async () => {
  // render(<App />);
  // const linkElement = screen.getByText(/Sound Recorder/i);
  // expect(linkElement).toBeInTheDocument();
  const component = await renderer.create(
    <App />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
