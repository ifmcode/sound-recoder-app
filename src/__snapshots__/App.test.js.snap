// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Sound Recorder title 1`] = `
Array [
  <header>
    <h1>
      Web Dictaphone
    </h1>
  </header>,
  <main>
    <p
      className="loading-phrase"
    >
      Loadingâ€¦
    </p>
  </main>,
]
`;
